File: ts/260.ts
function singleNumber(nums: number[]): number[] {
    let xor = nums.reduce((a, b) => a ^ b);
    let mask = xor & (-xor);
    let ans = [0, 0];
    for (let num of nums) {
        if ((num & mask) === 0) {
            ans[0] ^= num;
        } else {
            ans[1] ^= num;
        }
    }

    return ans;
};

File: kt/260.kt
class Solution {
    fun singleNumber(nums: IntArray): IntArray {
        val xorSum = nums.reduce { acc, i -> acc xor i }
        val lowbit = xorSum and -xorSum
        val ans = intArrayOf(0, 0)
        nums.forEach {
            if (it and lowbit == 0) {
                ans[0] = ans[0] xor it
            } else {
                ans[1] = ans[1] xor it
            }
        }

        return ans
    }
}